msal_extensions/persistence.py

```diff
@@ -41,6 +45,15 @@ def _mkdir_p(path):
             raise
 
 
+# We do not aim to wrap every os-specific exception.
+# Here we define only the most common one,
+# otherwise caller would need to catch os-specific persistence exceptions.
+class PersistenceNotFound(OSError):
```

Comment:
I am still thinking about `PersistenceNotFound` as an appropriate name of this error. Especially because right now it only caters to file not being found error. If there is a need in the future to address anything other than file not found, then this name would cause an issue. How about having a generic `PersistenceError`  instead ?

Reply:
After reconsideration, I still think `PersistenceNotFound` (or perhaps we can change it to `PersistenceNotExist`) is the right meaning we want to convey. The current name was chosen, NOT because we "cater to" the underlying FileNotFound error, but because that is the right semantics the higher level wants, i.e. our cache layer can confidently treat a "Persistence Not Found" as "cache is empty; business as usual".

Should we have a need in the future to address anything other than the underlying file-not-found, they are unlikely able to be safely ignored. So, they would NOT go into this error anyway.

By the way, that was also why the leading comments in line 48 says, "we do not aim to wrap everything" into this exception. Therefore, a more generic `PersistenceError` is NOT what we want here.

Reply:
I was thinking of a case where the persistence is corrupted and we want to handle such a case. Or even the case where file is empty but it does exist. I am concerned of the cases where we are expected to recover from or would be a no-op. I guess my question is why do we not aim to wrap everything in this exception ? So that we dont need a new one if any more cases need to be handled. Also we can always check this at the token_cache layer with an err_no like we did before to make is a no-op.

Reply:
Short answer is, if the persistence is corrupted, this kind of error will and should be bubbled up as a different exception, which will unlikely be treated as no-op. We literally do NOT aim to wrap everything here. This exception is just one "leaf" of a future exception hierarchy.

Reply:
But if we are to bubble it up as a different exception later, will it be ok adding such an exception later as persistence layer is public  API ?

Reply:
After discussion, we agreed it is OK to only have this `PersistenceNotFound` here, and it will be OK in the future if we wrap more "not previously-catchable" underlying platform specific exceptions into a meaningful new exception.